modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/Configuration.java

```diff
@@ -379,19 +379,8 @@ public int compare(DepNode o1, DepNode o2) {
      * @since 8.0
      */
     public <C extends Configuration> C getDependency(Class<C> clazz) {
-        //first try straight dependencies
-        for (Configuration dep : (List<Configuration>)getDependencies()) {
-            if (clazz.isInstance(dep)) {
-                return (C) dep;
-            }
-        }
-        //fall back to all
-        for (Configuration dep : (List<Configuration>)allDependencies()) {
-            if (clazz.isInstance(dep)) {
-                return (C) dep;
-            }
-        }
-        return null;
+        List dependencies = allDependencies();
+        return (C) dependencies.stream().filter(dep -> clazz.isInstance(dep)).findFirst().orElse(null);
```

Comment:
Does this change the order of dependency resolution?

The local variable "dependencies" is only used once and could be eliminated.

Reply:
I know... but I often like to have those around as debugging aids... otherwise the thing is computed and returned, maybe ending in another method call and return, and one has to do a few steps to actually see what the results was.

Reply:
About the order of dependency resolution, it may, but it's a consistency thing. WFS has two dependencies to GML, this method was returning one, and was being used by the GML output format to configure the decimals, and the method configuring the bindings was calling on "allDepenencies" instead, the result is that one GML configuration had the decimails set, but was not used. With this change all methods follow the same order and pick the same of the two GML dependencies.